forest/bonsai/simulate_gps_data.py

```diff
@@ -1117,31 +1122,310 @@ def process_switches(
+        try:
+            r = api.query(overpy_query)
+            break
+        except (
+            overpy.exception.OverpassTooManyRequests,
+            overpy.exception.OverpassGatewayTimeout
+        ):
+            time.sleep(30)
+
+    try:
+        index = np.random.choice(range(len(r.nodes)), 100, replace=False)
+    except ValueError:
+        raise ValueError(
+            "Overpass query came back empty. Check the location argument, ISO code and city name, for any misspellings."
+            )
+
+    nodes = np.array(r.nodes)[index]
+
+    return nodes
+
+
+def generate_nodes(
+    api: overpy.Overpass,
+    house_address: Tuple[float, float],
+    employment: Occupation
+) -> Dict[str, List[Tuple[float, float]]]:
+    """Generates multiple amenities coordinates.
+
+    Args:
+        api: (overpy.Overpass) overpass api
+        house_address: (tuple) coordinates of the house
+        employment: (Occupation) occupation of the person
+    Returns:
+        nodes: (dictionary) contains coordinates of each amenity
+    Raises:
+        RuntimeError: if the api raises error for too many tries
+    """
+
+    house_area = bounding_box(house_address, 2000)
+    house_area2 = bounding_box(house_address, 3000)
+
+    q_employment = ""
+    if employment == Occupation.WORK:
+        q_employment = f'node{house_area}["office"];'
+    elif employment == Occupation.SCHOOL:
+        q_employment = f'node{house_area2}["amenity"="university"];\n\t\t\tway{house_area2}["amenity"="university"]'
+
+    overpy_query2 = f"""
+    [out:json];
+    (
+        node{house_area}["amenity"="cafe"];
+        node{house_area}["amenity"="bar"];
+        node{house_area}["amenity"="restaurant"];
+        node{house_area}["amenity"="cinema"];
+        node{house_area}["leisure"="park"];
+        node{house_area}["leisure"="dance"];
+        node{house_area}["leisure"="fitness_centre"];
+        way{house_area}["amenity"="cafe"];
+        way{house_area}["amenity"="bar"];
+        way{house_area}["amenity"="restaurant"];
+        way{house_area}["amenity"="cinema"];
+        way{house_area}["leisure"="park"];
+        way{house_area}["leisure"="dance"];
+        way{house_area}["leisure"="fitness_centre"];
+        {q_employment}
+    );
+    out center;
+    """
+
+    for try_no in range(4):
+        if try_no == 3:
+            raise RuntimeError(
+                "Too many Overpass requests in a short time. Please try again in a minute."
+                )
+        try:
+            r = api.query(overpy_query2)
+            break
+        except (
+            overpy.exception.OverpassTooManyRequests,
+            overpy.exception.OverpassGatewayTimeout
+        ):
+            time.sleep(60)
+
+    all_nodes: Dict[str, list] = {}
+    for place in list(PossibleExits):
+        all_nodes[place.value] = []
+    all_nodes["office"] = []
+    all_nodes["university"] = []
+
+    for element in r.nodes + r.ways:
+        if element._type_value == "node":
```

Comment:
It's best not to use protected class members. I think `isinstance()` is appropriate here.
https://docs.python.org/3/library/functions.html#isinstance